MultiFacetLucene/FacetSearcher.cs

```diff
@@ -92,10 +98,24 @@ private IEnumerable<FacetMatch> FindMatchesInQuery(Query baseQueryWithoutFacetDr
 		{
 			var calculations = 0;
 			var queryFilter = new CachingWrapperFilter(CombineQueryWithFilter(CreateFacetedQuery(baseQueryWithoutFacetDrilldown, allFacetFieldInfos, facetFieldInfoToCalculateFor.FieldName), filter));
-			var bitsQueryWithoutFacetDrilldown = new OpenBitSetDISI(queryFilter.GetDocIdSet(IndexReader).Iterator(), IndexReader.MaxDoc);
+						
+			var bitsQueryWithoutFacetDrilldown = new OpenBitSetDISI(IndexReader.MaxDoc);
+			foreach (var leaf in IndexReader.Leaves)
+            {
+                var docSet = queryFilter.GetDocIdSet(leaf.AtomicReader.AtomicContext, leaf.AtomicReader.LiveDocs);
+                if (docSet == null)
+                {
+					continue;
+                }
+				var iterator = docSet.GetIterator();
+                if (iterator == null)
+                {
+                    continue;
+                }
+                bitsQueryWithoutFacetDrilldown.InPlaceOr(iterator);
```

Comment:
Formatting has gone a bit astray here
```suggestion
                {
                    var docSet = queryFilter.GetDocIdSet(leaf.AtomicReader.AtomicContext, leaf.AtomicReader.LiveDocs);
                    if (docSet == null)
                    {
		    			continue;
                    }
	    			var iterator = docSet.GetIterator();
                    if (iterator == null)
                    {
                        continue;
                    }
                    bitsQueryWithoutFacetDrilldown.InPlaceOr(iterator);
```